### DISCUSSION TITLE
開発環境について

### DISCUSSION THREAD

**Discussion by taniiicom** [gdsc-tmu/InfoLinkAPI#23] [General]
これについて, 自分もどうするか, ちょうど考えてたポイントでした.  
これについては, いろんな思想があると思いますが, 個人的にバランスの取れている良い開発環境だと思っているものを以下に書きます.  
- ローカルの環境構築はしんどいので, docker を使いたい. ただ, 開発体験を改善するために以下のようにする.  
- まず, docker のビルド時にテストを実行するのではなく, テスト部分を分離して, docker の build 時の処理を必要最低限にする
- テストは, docker exec でコンテナ内に入って, go test を実行する. 
- ただ, これを毎回するのは大変なので, test.sh のようなテスト実行スクリプトを作成し, それを実行すれば, 自動で docker exec して, コンテナの中でテストが実行できるようにする
- これで, docker 自体は, linux の上に, go を載せて, その上で build するだけのシンプルな責務になったので, Air を用いて, ホットリロードの環境を作る
- Air によって, コードに変更が生じた場合は, 自動的にリビルドされ, 変更が反映される

この環境の利点というか, モチベーションを以下に書きます.  
- エラーが発生しうるコードでもとりあえず build には成功して, 実行できることで問題が特定できる
- os や環境によって生じるエラーと, アプリケーションコードによって生じているエラーは明確に分離する
- コードの変更後の実行にかかる時間を短縮する

あと, もし可能なら, Slack は90日で投稿が見れなくなってしまうのと, 1 つの場所に情報を残しておくという意味でも, 
issue ほど明確な問題ではないけど, 議論が生じうる議題については, github の discussion に移動するのはどうでしょうか?

https://gdsctokyometr-x4d8844.slack.com/archives/C06C2TLS8EM/p1708566708469249

**Comment by Atotti** [comment]
よさそう

**Comment by tenk-9** [comment]
環境とdiscussion共にとても良いと思います
